src/prompt/variable.rs

```diff
@@ -186,43 +536,57 @@ impl Hinter for VarHelper {
         if line == "" {  // be friendly from the start
             return None;
         }
-        match self.var_type.parse(line) {
-            Ok(_) => None,
-            Err(Error::Incomplete) => None,
-            Err(e) => Some(ErrorHint(format!(" -- {}", e))),
+        match self.var_type.parse(line, InputFlags::None) {
+            Ok(r) => {
+                if r.0.len() == 0 {
+                    return None
+                }
+
+                return Some(ErrorHint(" -- excess unparsed content".to_string()))
+            },
+            Err(e) => {
+                Some(ErrorHint(format_parsing_error(e)))
+            }
         }
     }
 }
 
 impl Highlighter for VarHelper {
     fn highlight<'l>(&self, line: &'l str, _pos: usize) -> Cow<'l, str> {
-        match self.var_type.parse(line) {
-            Ok(_) => line.into(),
+        match self.var_type.parse(line, InputFlags::None) {
+            Ok(r) => {
+                if r.0.len() == 0 {
+                    return line.into()
+                }
+
+                let mut str = line[..(line.len() - r.0.len())].to_string();
```

Comment:
explain?

Reply:
slices the original input to remove the remaining unparsed content from it, to then color the remaining red

Reply:
put that in a comment please